backend/corpora/lambdas/api/v1/dataset.py

```diff
@@ -1,8 +1,15 @@
 from flask import make_response, jsonify
 
+from ....common.corpora_orm import DbDatasetProcessingStatus, UploadStatus
 from ....common.entities import Dataset, Collection
 from ....common.utils.db_utils import db_session
-from ....common.utils.exceptions import NotFoundHTTPException, ServerErrorHTTPException, ForbiddenHTTPException
+from ....common.utils.exceptions import (
+    NotFoundHTTPException,
+    ServerErrorHTTPException,
+    ForbiddenHTTPException,
+    MethodNotAllowedException,
+)
+from ....dataset_processing.download import processing_status_updater
```

Comment:
I'd avoid importing from data_processing, we might start running into weird circular dependencies in the future if we do. You can access and update the `upload_status` through dataset.

`processing_status_updater` was written as an optimized way of updating the upload_status since we were modifying it frequently. It wasn't supposed to really be used outside of the downloader code.

To modify the upload_status you can do something like this 
```python
from corpora.common.entities.dataset import Dataset
from corpora.common.corpora_orm import UploadStatus
from corpora.common.utils.db_utils import db_session_manager

with db_session_manager(commit=True):
  data_uuid = "test_dataset_id"
  dataset = Dataset.get(dataset_uuid)
  dataset.processing_status.upload_status = UploadStatus.CANCEL_PENDING
```

Reply:
discussed over slack
-> I am going to move the processing_status_updater into db_utils